services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/ApplicationConfig.java

```diff
@@ -57,19 +59,35 @@ public Vertx vertx() {
      * 
      * @return The factory.
      */
+    @Qualifier("amqp")
     @Bean
-    public ObjectFactoryCreatingFactoryBean deviceRegistryServerFactory() {
+    public ObjectFactoryCreatingFactoryBean amqpServerFactory() {
         ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
         factory.setTargetBeanName(DeviceRegistryAmqpServer.class.getName());
         return factory;
     }
 
     /**
+     * FIXME
+     * Exposes a factory for creating service instances as a Spring bean.
+     *
+     * @return The factory.
+     */
+    @Qualifier("rest")
+    @Bean
+    public ObjectFactoryCreatingFactoryBean restServerFactory() {
+        ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
+        factory.setTargetBeanName(DeviceRegistryRestServer.class.getName());
+        return factory;
+    }
+
+    /**
      * Exposes this service's AMQP endpoint configuration properties as a Spring bean.
      * 
      * @return The properties.
      */
     @Qualifier("amqp")
+    @Primary // needed for class BaseEndpoint
```

Comment:
`BaseEndpoint.setConfiguration()` can not autowire these `ServiceConfigProperties` because a bean of the same type exists for REST. Alternative would be hardcoding `@Qualifier("amqp")` in `BaseEndpoint`.